src/main/java/org/icgc_argo/workflow/relay/service/IndexService.java

```diff
@@ -43,28 +46,41 @@ public IndexService(
   @SneakyThrows
   @StreamListener(IndexStream.WORKFLOW)
   public void indexWorkflow(JsonNode event) {
-    // TODO: Generalize event information parsing
+    // deserialize json events to metadata objects
+    val tempMapper =
+        new ObjectMapper().configure(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES, false);
```

Comment:
you can use jackson annotations to ignore unknown properties and use the same mapper